### PR TITLE
VM OrchestrationMode: PlatformFaultDomainCount in New-AzVmss and VmssId in New-AzVm

### PR DESCRIPTION
**Pull request by Sandido**
## Checklist

- [x] I have read the [_Submitting Changes_](../blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/master/CONTRIBUTING.md)
- [x] The title of the PR is clear and informative
- [x] The appropriate `ChangeLog.md` file(s) has been updated:
    - For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    - A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header -- no new version header should be added
- [ ] The PR does not introduce [breaking changes](../blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [x] If applicable, the changes made in the PR have proper test coverage
- [x] For public API changes to cmdlets:
    - [x] a cmdlet design review was approved for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (_Microsoft internal only_)
    - [ ] the markdown help files have been regenerated using the commands listed [here](../blob/master/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)

### CODE REVIEW

File: src/Compute/Compute/Generated/VirtualMachineScaleSet/Config/NewAzureRmVmssConfigCommand.cs [hunk 1]
```diff
@@ -363,7 +363,21 @@ private void Run()
...
+                if (vUpgradePolicy.AutomaticOSUpgradePolicy == null)
+                {
+                    vUpgradePolicy.AutomaticOSUpgradePolicy = new AutomaticOSUpgradePolicy();
+                }
+                vUpgradePolicy.AutomaticOSUpgradePolicy.EnableAutomaticOSUpgrade = this.AutoOSUpgrade.IsPresent;
+            }
 
+            /*
             if (vUpgradePolicy == null)
             {
```
Comment (Sandido) [inline src/Compute/Compute/Generated/VirtualMachineScaleSet/Config/NewAzureRmVmssConfigCommand.cs:382]: Need to remove this comment.

**Comment by msJinLei** [comment]
@Sandido  could you rebase to latest the master to suppress the test timeout?

**Review by msJinLei — APPROVED** [review]

File: src/Compute/Compute.Test/ScenarioTests/VirtualMachineScaleSetTests.ps1 [hunk 2]
```diff
@@ -2537,6 +2537,73 @@ function Test-VirtualMachineScaleSetEncryptionAtHost
...
+        $cred = New-Object System.Management.Automation.PSCredential ($adminUsername, $securePassword);
+
+
+        $VmssConfigWithoutVmProfile = new-azvmssconfig -location $loc -platformfaultdomain 1
+        $VmssWithoutVmProfile = new-azvmss -resourcegroupname $rgname -vmscalesetname $vmssName -virtualmachinescaleset $VmssConfigWithoutVmProfile
+
+        $vm = new-azvm -resourcegroupname $rgname -location $loc -name $vmname -credential $cred -domainnamelabel $domainName -vmssid $VmssWithoutVmProfile.id
+
+        Assert-AreEqual $VmssWithoutVmProfile.id $vm.virtualmachinescaleset.id
+
```
Comment (grizzlytheodore) [inline src/Compute/Compute.Test/ScenarioTests/VirtualMachineScaleSetTests.ps1:2572]: is there a way to view the instance view of vmss to see this vm added to the vmss?

File: src/Compute/Compute/Manual/VirtualMachineScaleSetCreateOrUpdateMethod.cs [hunk 3]
```diff
@@ -180,6 +180,11 @@ public partial class NewAzureRmVmss : ComputeAutomationBaseCmdlet
         [Parameter(ParameterSetName = SimpleParameterSet, Mandatory = false)]
         public SwitchParameter EncryptionAtHost { get; set; }
 
+        [Parameter(ParameterSetName = SimpleParameterSet, Mandatory = false,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "Fault Domain count for each placement group.")]
+        public int? PlatformFaultDomainCount { get; set; }
```
Comment (grizzlytheodore) [inline src/Compute/Compute/Manual/VirtualMachineScaleSetCreateOrUpdateMethod.cs:186]: i believe '?' is not necessary here to be able to pass this parameter as null.  Should behave the same when the variable type is 'int',   check 'maxprice' parameter. I

File: src/Compute/Compute/VirtualMachine/Operation/NewAzureVMCommand.cs [hunk 4]
```diff
@@ -233,6 +233,10 @@ public class NewAzureVMCommand : VirtualMachineBaseCmdlet
         [Parameter(ParameterSetName = SimpleParameterSet, Mandatory = false)]
         [Parameter(ParameterSetName = DiskFileParameterSet, Mandatory = false)]
         public string HostId { get; set; }
+        
+        [Parameter(ParameterSetName = SimpleParameterSet, Mandatory = false)]
+        [Parameter(ParameterSetName = DiskFileParameterSet, Mandatory = false)]
+        public string VmssId { get; set; }
```
Comment (grizzlytheodore) [inline src/Compute/Compute/VirtualMachine/Operation/NewAzureVMCommand.cs:240]: should we add vmssId parameter to new-azvmConfig?

File: src/Compute/Compute/Manual/VirtualMachineScaleSetCreateOrUpdateMethod.cs (hunk 3 above)
Comment (Sandido) [reply src/Compute/Compute/Manual/VirtualMachineScaleSetCreateOrUpdateMethod.cs:186]: @grizzlytheodore , having the parameter as the nullable int type (int?) removes the need to cast it as a nullable int in the resourceGroup.CreateVirtualMachineScaleSetConfig() call. 
MaxPrice does have this cast (double?), so having the parameter int? removes the need for this cast.

File: src/Compute/Compute/VirtualMachine/Operation/NewAzureVMCommand.cs (hunk 4 above)
Comment (Sandido) [reply src/Compute/Compute/VirtualMachine/Operation/NewAzureVMCommand.cs:240]: @grizzlytheodore , it is already there, [the -vmssId parameter.](https://docs.microsoft.com/en-us/powershell/module/az.compute/new-azvmconfig?view=azps-4.7.0)

File: src/Compute/Compute/Manual/VirtualMachineScaleSetCreateOrUpdateMethod.cs (hunk 3 above)
Comment (grizzlytheodore) [reply src/Compute/Compute/Manual/VirtualMachineScaleSetCreateOrUpdateMethod.cs:186]: @Sandido  I see. I think it's better to stay consistent on where the parameter is being casted as a nullable variable. Not only in this cmdlet, the variables are casted as a nullable variable in CreateVirtualMachineScaleSetConfig(), but other cmdlets follow the similar structure where '?' is left out at the parameter declaration, and later is casted as nullable when it is being checked with isParameter() method.

Comment (Sandido) [reply src/Compute/Compute/Manual/VirtualMachineScaleSetCreateOrUpdateMethod.cs]: @grizzlytheodore , I changed it as you suggested.

File: src/Compute/Compute.Test/ScenarioTests/VirtualMachineScaleSetTests.ps1 (hunk 2 above)
Comment (Sandido) [reply src/Compute/Compute.Test/ScenarioTests/VirtualMachineScaleSetTests.ps1:2572]: @grizzlytheodore , No. This functionality does not exist in PS and CLI due to how the feature is setup on the server side.

**Review by markcowl — APPROVED** [review]

**Pull request merged**